src/main/java/com/flagsmith/mappers/EngineMappers.java

```diff
@@ -0,0 +1,387 @@
+      JsonNode featureStates) {
+
+    List<FeatureContext> featureContexts = new ArrayList<>();
+
+    for (JsonNode featureState : featureStates) {
+      FeatureContext featureContext = mapFeatureStateToFeatureContext(featureState);
+      featureContexts.add(featureContext);
+    }
+
+    return featureContexts;
+  }
+
+  /**
+   * Gets the feature state key from either django_id or featurestate_uuid.
+   *
+   * @param featureState the feature state JSON
+   * @return the feature state key as string
+   */
+  private static String getFeatureStateKey(JsonNode featureState) {
+    JsonNode node = featureState.get("django_id");
+    if (node != null && !node.isNull()) {
+      return node.asText();
+    }
+    node = featureState.get("featurestate_uuid");
+    if (node != null && !node.isNull()) {
+      return node.asText();
+    }
+    return "";
+  }
+
+  private static Object getFeatureStateValue(JsonNode featureState, String fieldName) {
+    JsonNode valueNode = featureState.get(fieldName);
+    if (valueNode.isTextual() || valueNode.isLong()) {
+      return valueNode.asText();
+    } else if (valueNode.isNumber()) {
+      if (valueNode.isInt()) {
+        return valueNode.asInt();
+      } else {
+        return valueNode.asDouble();
+      }
+    } else if (valueNode.isBoolean()) {
+      return valueNode.asBoolean();
+    } else if (valueNode.isArray() || valueNode.isObject()) {
+      return valueNode;
+    }
+    return null;
+  }
+
+  /**
+   * Maps a single feature state to feature context.
+   *
+   * @param featureState the feature state JSON
+   * @return the feature context
+   */
+  private static FeatureContext mapFeatureStateToFeatureContext(JsonNode featureState) {
+    JsonNode feature = featureState.get("feature");
+
+    FeatureContext featureContext = new FeatureContext()
+        .withKey(getFeatureStateKey(featureState))
+        .withFeatureKey(feature.get("id").asText())
+        .withName(feature.get("name").asText())
+        .withEnabled(featureState.get("enabled").asBoolean())
+        .withValue(getFeatureStateValue(featureState, "feature_state_value"));
+
+    // Handle multivariate feature state values
+    JsonNode multivariateValues = featureState.get("multivariate_feature_state_values");
+    if (multivariateValues != null && multivariateValues.isArray()) {
+      List<FeatureValue> variants = new ArrayList<>();
+      List<JsonNode> sortedMultivariate = new ArrayList<>();
+      multivariateValues.forEach(sortedMultivariate::add);
+      sortedMultivariate.sort((a, b) -> a.get("id").asText().compareTo(b.get("id").asText()));
+      for (JsonNode multivariateValue : sortedMultivariate) {
+        FeatureValue variant = new FeatureValue()
+            .withValue(getFeatureStateValue(
+              multivariateValue.get("multivariate_feature_option"), "value"))
+            .withWeight(multivariateValue.get("percentage_allocation").asDouble());
+        variants.add(variant);
```

Comment:
Keeping on the radar. It seems we implement `variants = []` for single value features but the schema is saying the opposite:
```
                "variants": {
                    "description": "An array of environment default values associated with the feature. Contains a single value for standard features, or multiple values for multivariate features.",
                    "items": {
                        "$ref": "#/$defs/FeatureValue"
                    },
                    "title": "Variants",
                    "type": "array"
                },
```